src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1821,7 +1864,22 @@ public static boolean fromTheSameCaseExpression(@NotNull PsiElement origin, @Not
     return false;
   }
 
-  public static class ErlangFunctionCallParameter<T extends PsiElement> extends PatternCondition<T> {
+  public static abstract class ErlangFunctionCallParameterBase<T extends PsiElement> extends PatternCondition<T> {
+    protected ErlangFunctionCallExpression myCall;
+    protected int myPositionInCall;
```

Comment:
Avoid storing state in patterns! What if your pattern is used concurrently? It will be a mess.